server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/ResponseSerializer.scala

```diff
@@ -225,6 +225,13 @@ object ResponseSerializer {
     case error: WebSocketError => webSocketErrorWrites.writes(error)
   }
 
+  private implicit val typeNameReads: Reads[TypeName] = {
+    case JsString(value) if value.equals(MailboxTypeName.asString()) => JsSuccess(MailboxTypeName)
+    case JsString(value) if value.equals(EmailTypeName.asString()) => JsSuccess(EmailTypeName)
+    case _ => JsError("Expecting a JsString as typeName")
+  }
+  private implicit val webSocketPushEnableReads: Reads[WebSocketPushEnable] = Json.reads[WebSocketPushEnable]
```

Comment:
Then we need to handle this within the WebsocketInbondMessage deserializer

Reply:
resolved by your commit?